Use enumToUnderlyingType more in WebCore

fujii opened the pull request:
#### 776f4b7a841cc8c8f26c276d0ae23230caed54ef
<pre>
Use enumToUnderlyingType more in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=266173">https://bugs.webkit.org/show_bug.cgi?id=266173</a>

Reviewed by Chris Dumez.

* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
* Source/WebCore/animation/AcceleratedTimeline.cpp:
* Source/WebCore/bindings/js/JSDOMMapLike.cpp:
* Source/WebCore/bindings/js/JSDOMSetLike.cpp:
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
* Source/WebCore/bindings/js/JSLocalDOMWindowCustom.cpp:
* Source/WebCore/bindings/js/JSObservableArray.cpp:
* Source/WebCore/bindings/js/JSPluginElementFunctions.cpp:
* Source/WebCore/bindings/js/JSRemoteDOMWindowCustom.cpp:
* Source/WebCore/bindings/js/ScriptModuleLoader.cpp:
* Source/WebCore/bridge/runtime_array.cpp:
* Source/WebCore/bridge/runtime_object.cpp:
* Source/WebCore/crypto/CryptoAlgorithmRegistry.cpp:
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSRule.cpp:
* Source/WebCore/css/CSSSelector.cpp:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/CSSValuePool.cpp:
* Source/WebCore/css/DeprecatedCSSOMValue.h:
* Source/WebCore/css/ImmutableStyleProperties.cpp:
* Source/WebCore/css/MutableStyleProperties.cpp:
* Source/WebCore/css/calc/CSSCalcOperationNode.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentMarkerController.cpp:
* Source/WebCore/dom/Element.cpp:
* Source/WebCore/dom/EventTarget.h:
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/Node.h:
* Source/WebCore/dom/make_names.pl:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLInputElement.h:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/platform/graphics/FontCascadeDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.h:
* Source/WebCore/rendering/RenderGrid.cpp:
* Source/WebCore/rendering/style/NinePieceImage.h:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/StyleContentAlignmentData.h:
* Source/WebCore/rendering/style/StyleSelfAlignmentData.h:
* Source/WebCore/style/RuleFeature.h:
* Source/WebCore/style/StyleScopeOrdinal.h:
* Source/WebCore/svg/SVGPathByteStreamBuilder.h:
* Source/WebCore/workers/service/server/SWRegistrationDatabase.cpp:

Canonical link: <a href="https://commits.webkit.org/272145@main">https://commits.webkit.org/272145@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/01a524cdac80741ddaf82c9d72cbf9123b8ce1a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30592 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9274 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32246 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33087 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27657 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31296 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11560 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6524 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27567 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30900 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7773 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27388 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6656 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6814 "Too many flaky failures: imported/w3c/web-platform-tests/css/css-cascade/scope-hover.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-partially-onscreen-new.html, imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-below-viewport-partially-onscreen-old.html, imported/w3c/web-platform-tests/css/cssom-view/scroll-behavior-main-frame-window.html, imported/w3c/web-platform-tests/css/selectors/invalidation/user-valid-user-invalid.html, imported/w3c/web-platform-tests/css/selectors/user-invalid.html, imported/w3c/web-platform-tests/css/selectors/valid-invalid-form-fieldset.html, imported/w3c/web-platform-tests/feature-policy/feature-policy-frame-policy-allowed-for-all.https.sub.html, imported/w3c/web-platform-tests/html/semantics/disabled-elements/disabled-event-dispatch.tentative.html, imported/w3c/web-platform-tests/html/semantics/invokers/invoketarget-button-event-dispatch.tentative.html, imported/w3c/web-platform-tests/html/semantics/selectors/pseudo-classes/readwrite-readonly.html (failure)") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27246 "Found 1 new test failure: svg/compositing/transform-change-repainting-viewBox.html (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34423 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27844 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27742 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32982 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6833 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4941 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30808 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8566 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7265 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7561 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7390 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/dom/Document.h:
```diff
@@ -321,7 +321,7 @@ enum class NodeListInvalidationType : uint8_t {
     InvalidateOnAnyAttrChange,
 };
-const uint8_t numNodeListInvalidationTypes = static_cast<uint8_t>(NodeListInvalidationType::InvalidateOnAnyAttrChange) + 1;
+const uint8_t numNodeListInvalidationTypes = enumToUnderlyingType(NodeListInvalidationType::InvalidateOnAnyAttrChange) + 1;
```
would be nice to use `std::underlying_type_t<NodeListInvalidationType>` instead of `uint8_t` for the variable then, or `auto`

fujii replied:
Fixed. Uses `auto'.

cdumez commented on Source/WebCore/dom/Document.h:
```diff
@@ -618,7 +618,7 @@ class Document
 
     static ptrdiff_t documentClassesMemoryOffset() { return OBJECT_OFFSETOF(Document, m_documentClasses); }
-    static uint32_t isHTMLDocumentClassFlag() { return static_cast<uint32_t>(DocumentClass::HTML); }
+    static uint32_t isHTMLDocumentClassFlag() { return enumToUnderlyingType(DocumentClass::HTML); }
```
ditto about `std::underlying_type_t` or `auto`

cdumez commented on Source/WebCore/dom/Node.h:
```diff
@@ -571,13 +571,13 @@ class Node : public EventTarget {
-    static int32_t flagIsHTML() { return static_cast<int32_t>(NodeFlag::IsHTMLElement); }
-    static int32_t flagIsLink() { return static_cast<int32_t>(NodeFlag::IsLink); }
-    static int32_t flagIsParsingChildrenFinished() { return static_cast<int32_t>(NodeFlag::IsParsingChildrenFinished); }
+    static int32_t flagIsText() { return enumToUnderlyingType(NodeFlag::IsText); }
```
ditto.

fujii replied:
Fixed

fujii committed:
Use enumToUnderlyingType more in WebCore
https://bugs.webkit.org/show_bug.cgi?id=266173

Reviewed by Chris Dumez.

* Source/WebCore/accessibility/AXObjectCache.cpp:
* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
* Source/WebCore/animation/AcceleratedTimeline.cpp:
* Source/WebCore/bindings/js/JSDOMMapLike.cpp:
* Source/WebCore/bindings/js/JSDOMSetLike.cpp:
* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
* Source/WebCore/bindings/js/JSDOMWindowProperties.cpp:
* Source/WebCore/bindings/js/JSLocalDOMWindowCustom.cpp:
* Source/WebCore/bindings/js/JSObservableArray.cpp:
* Source/WebCore/bindings/js/JSPluginElementFunctions.cpp:
* Source/WebCore/bindings/js/JSRemoteDOMWindowCustom.cpp:
* Source/WebCore/bindings/js/ScriptModuleLoader.cpp:
* Source/WebCore/bridge/runtime_array.cpp:
* Source/WebCore/bridge/runtime_object.cpp:
* Source/WebCore/crypto/CryptoAlgorithmRegistry.cpp:
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSRule.cpp:
* Source/WebCore/css/CSSSelector.cpp:
* Source/WebCore/css/CSSSelector.h:
* Source/WebCore/css/CSSValuePool.cpp:
* Source/WebCore/css/DeprecatedCSSOMValue.h:
* Source/WebCore/css/ImmutableStyleProperties.cpp:
* Source/WebCore/css/MutableStyleProperties.cpp:
* Source/WebCore/css/calc/CSSCalcOperationNode.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/dom/DocumentMarkerController.cpp:
* Source/WebCore/dom/Element.cpp:
* Source/WebCore/dom/EventTarget.h:
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/Node.h:
* Source/WebCore/dom/make_names.pl:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLInputElement.h:
* Source/WebCore/inspector/agents/InspectorDOMAgent.cpp:
* Source/WebCore/platform/graphics/FontCascadeDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.h:
* Source/WebCore/rendering/RenderGrid.cpp:
* Source/WebCore/rendering/style/NinePieceImage.h:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/StyleContentAlignmentData.h:
* Source/WebCore/rendering/style/StyleSelfAlignmentData.h:
* Source/WebCore/style/RuleFeature.h:
* Source/WebCore/style/StyleScopeOrdinal.h:
* Source/WebCore/svg/SVGPathByteStreamBuilder.h:
* Source/WebCore/workers/service/server/SWRegistrationDatabase.cpp:

Canonical link: https://commits.webkit.org/272145@main
